src/pq_flash_index.cpp

```diff
@@ -932,14 +936,21 @@ namespace diskann {
       }
     }
 
-    compute_dists(&best_medoid, 1, dist_scratch);
+    std::vector<_u32> init_nodes;
+    init_nodes.reserve(1);
+    init_nodes.emplace_back(best_medoid);
+    visited.insert(best_medoid);
+    add_extra_start_points(query1, l_search, visited, init_nodes);
+
+    compute_dists(init_nodes.data(), init_nodes.size(), dist_scratch);
     /*retset[0].id = best_medoid;
     retset[0].distance = dist_scratch[0];
     retset[0].flag = true;*/
-    retset.push_back(Neighbor(best_medoid, dist_scratch[0], true));
-    visited.insert(best_medoid);
```

Comment:
please check if the space allotted for retset is sufficient to push all the init_nodes (after rebasing to main)